src/components/LottoGame.js

```diff
@@ -0,0 +1,113 @@
+import PurchaseForm from "./PurchaseForm.js";
+import LottoPurchaseHistory from "./LottoPurchaseHistory.js";
+import LottoWinningInfoForm from "./LottoWinningInfoForm.js";
+import WinningStatistic from "./WinningStatistic.js";
+import Modal from "./common/Modal.js";
+import LottoMachine from "../domain/LottoMachine.js";
+import LottoResult from "../domain/LottoResult.js";
+import lottoTransactionStore from "../store/lottoTransactionStore.js";
+import winningLottoInfoStore from "../store/winningLottoInfo.js";
+import customCreateElement from "../utils/customElement.js";
+
+export default class LottoGame {
+  #target;
+  #show;
+  $div;
+
+  constructor($target) {
+    this.#target = $target;
+    this.#show = false;
+
+    lottoTransactionStore.subscribe(() => this.render());
+    winningLottoInfoStore.subscribe(() => this.render());
+
+    this.renderStaticElement();
+    this.render();
+  }
+
+  setShow = (newState) => {
+    this.#show = newState;
+    this.render();
+  };
+
+  setInit = () => {
+    this.#show = false;
+
+    lottoTransactionStore.resetState();
+    winningLottoInfoStore.resetState();
+  };
+
+  renderStaticElement() {
+    const $div = customCreateElement({
+      tagName: "div",
+      className: "lotto-winning-result-container",
+    });
+    const $title = customCreateElement({
+      tagName: "div",
+      className: "lotto-winning-result-title",
+      text: "🎱 내 번호 당첨 확인 🎱",
+    });
+
+    $div.appendChild($title);
+    this.#target.appendChild($div);
+
+    new PurchaseForm($div, this.setShow);
+
+    this.$div = customCreateElement({
+      tagName: "div",
+    });
+
+    $div.appendChild(this.$div);
+  }
+
+  resetStaticElements() {
+    const $modal = document.querySelector(".modal-bg");
+    if ($modal) $modal.remove();
+
+    const $input = document.querySelector(".purchase-form-input");
+    $input.value = "";
+  }
+
+  render() {
+    this.resetStaticElements();
+
+    this.$div.replaceChildren();
+
+    const $modal = document.querySelector(".modal-bg");
+    if ($modal) $modal.remove();
+
+    const $input = document.querySelector(".purchase-form-input");
+    $input.value = "";
+
+    this.$div.replaceChildren();
+
+    new LottoPurchaseHistory(this.$div, this.#show);
+    new LottoWinningInfoForm(this.$div, this.#show);
+
+    const winningResult = this.calculateWinningResult();
+
+    if (!winningResult) return;
+
+    const winningStatistic = new WinningStatistic(winningResult, this.setInit);
+
+    new Modal(this.#target, ($target) => winningStatistic.render($target));
```

Comment:
모든 컴포넌트에서 target 파라미터를 직접 받는 이유가 궁금합니다~!
appendChild만 하는 것이 목적이라면, 요소를 반환받아서 직접 하는 것은 어떤가요?

```suggestion
    this.#target.appendChild(new Modal(winningStatistic.render()));
```

Reply:
$target을 넘겨주는 이유는 해당 컴포넌트를 렌더하고 싶은 곳이 달라질 수 있으니 확장성을 고려해서 $target을 받도록 했습니다. 그런데 이렇게 하면 붙여주는 곳을 알지 않아도 되어서 오히려 좋다고 생각됩니다! render를 "화면에 그린다" 는 생각만 해서 요소를 반환할 생각을 못했네요ㅎㅎ